TracyOpenCL.hpp

```diff
@@ -139,23 +148,33 @@ namespace tracy {
             }
 #endif
 
-            while (m_tail != m_head)
+            for (; m_tail != m_head; m_tail = (m_tail + 1) % QueryCount)
             {
-                EventInfo eventInfo = m_query[m_tail];
-                cl_event event = eventInfo.event;
+                EventInfo eventInfo = GetQuery(m_tail);
```

Comment:
I'm not sure it is worth going through the `GetQuery` function here. We are inside the `OpenCLCtx` and this loop is supposed to be very efficient in its good path. The range defined by `m_head` and `m_tail` is guaranteed to be valid when this is being executed.

Reply:
Previously, I wanted to add some 'streaming' functionality to it, for example, if the queue is full, then automatically `Collect` or something the like. But the idea is abandoned since I found without it, the code still works smoothly even when a large amount of events are queued. I think at least it does no harm and improve the readibility.